### PR TITLE
Implemented new 'unload' command in caravan script.

### PR DESCRIPTION
**Pull request by CarabusX**
Helps in cases of endless unloading at the depot.

### CODE REVIEW

File: caravan.lua [hunk 1]
```diff
@@ -95,6 +97,45 @@ function commands.leave(...)
...
+end
+
+local function getPrintableUnitName(unit)
+    local visible_name = dfhack.units.getVisibleName(unit)
+    local profession_name = dfhack.units.getProfessionName(unit)
+    if visible_name.has_name then
+        return dfhack.TranslateName( visible_name ) .. ' (' .. profession_name .. ')'
+    else
+        return profession_name  -- for unnamed animals
+    end
```
Comment (myk002) [inline caravan.lua:117]:
```suggestion
    if visible_name.has_name then
        return ('%s (%s)'):format(dfhack.TranslateName(visible_name), profession_name)
    end
    return profession_name  -- for unnamed animals
```

**Review by myk002 — CHANGES_REQUESTED** [review]
Code looks good! "rejoin_pack_animals" is kind of a long name, but it is specific and understandable. Calling it just "fix" is an option, and is "forward compatible" in case there are more things we can check for and fix in the future. @lethosor, what do you think?

Could you add a changelog entry in `changelog.txt` that describes this change? It should go in the "Misc Improvements" section of the "Future" release.

**Comment by myk002** [comment]
Also, please consider adding your name to the https://github.com/DFHack/dfhack/blob/develop/docs/Authors.rst file in the DFHack/dfhack repo.

**Comment by CarabusX** [comment]
> Code looks good! "rejoin_pack_animals" is kind of a long name, but it is specific and understandable. Calling it just "fix" is an option, and is "forward compatible" in case there are more things we can check for and fix in the future. @lethosor, what do you think?

Yeah name is kinda long, thats why I called it "rejoin" and not "reconnect".
In case there are more things to fix in the future, it may be good to have separate commands for that, so maybe better if current command is specific?

> Could you add a changelog entry in `changelog.txt` that describes this change? It should go in the "Misc Improvements" section of the "Future" release.

Sure! I will add it there.

**Comment by CarabusX** [comment]
> Also, please consider adding your name to the https://github.com/DFHack/dfhack/blob/develop/docs/Authors.rst file in the DFHack/dfhack repo.

Sure! I will add myself there later.

File: caravan.lua (hunk 1 above)
Comment (CarabusX) [reply caravan.lua:117]: I don't like this change as it would print " (Horse)", while I want to print units as they appear in unit list.

Comment (CarabusX) [reply caravan.lua:117]: I will test it and commit.

**Review by myk002 — APPROVED** [review]

**Review by lethosor — COMMENTED** [review]
I agree with @myk002 that the name is kind of long. I think it's also more specific than it needs to be. From what I understand, it took you some time to figure out that your caravan was stuck because the pack animals were disconnected. I don't think we can expect users to figure that out. From their perspective, the caravan is just stuck, and they don't know why, so I think `unstick` or something similar would be better (and shorter).

Now, if we want to maintain the ability to apply specific, targeted fixes in the future if `unstick` does more than one thing, I would be in favor of making `unstick()` call out to `rejoin_pack_animals()` and whatever else is necessary to get a caravan unstuck.

**Comment by CarabusX** [comment]
> I agree with @myk002 that the name is kind of long. I think it's also more specific than it needs to be. From what I understand, it took you some time to figure out that your caravan was stuck because the pack animals were disconnected. I don't think we can expect users to figure that out. From their perspective, the caravan is just stuck, and they don't know why, so I think `unstick` or something similar would be better (and shorter).
> 
> Now, if we want to maintain the ability to apply specific, targeted fixes in the future if `unstick` does more than one thing, I would be in favor of making `unstick()` call out to `rejoin_pack_animals()` and whatever else is necessary to get a caravan unstuck.

I will call it "unload" then, because it is only about caravan not unloading.
"unstick" seems too general - is the caravan stuck moving, or entering map, or leaving map, or unloading goods?

I will make `unload()` call out to `rejoin_pack_animals()` internally.

**Comment by myk002** [comment]
btw, your master branch will be out of sync after this is merged. you'll want to
```
git fetch upstream
git checkout master
git reset --hard upstream/master
git push --force
```
to get back in sync. Creating branches for PRs will avoid this issue in the future.

**Pull request merged**